apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/AbstractSpan.java

```diff
@@ -388,10 +438,19 @@ public void resetState() {
         userOutcome = null;
         hasCapturedExceptions = false;
         bufferedSpan.set(null);
+        recycleSpanLinks();
         otelKind = null;
         otelAttributes.clear();
     }
 
+    private void recycleSpanLinks() {
```

Comment:
This makes sense, but I think it makes more sense that extracting from the pool and recycling back are located together. Since this is a managed pool owned by the `tracer` and the logic for creating a span link fits here, I would leave it here and keep the list as "dumb" data structure as possible.